### PR TITLE
update manifests deployments for gpu-service

### PR DESCRIPTION
**Pull request by tybalex**
**what does this PR do:**
as the title implies. the `training-controller` service has been renamed to `gpu-service`, which is now a pod with 2 containers -- `gpu-service-controller` and `gpu-service-worker`. 
- `gpu-service-controller` merged from the old `training-controller`, basically manages both training and inference requests for gpu-service. 
- `gpu-service-worker` is the container that actually owns the GPU, and it executes any requests forward from the controller.

**Dependency:**
- [ ] https://github.com/rancher/opni-training-controller/pull/3
- [ ] https://github.com/rancher/opni-inference-service/pull/5

these 2 PRs should get merged before this PR. build is failing as the new images aren't existed yet.

### CODE REVIEW

**Review by dbason — APPROVED** [review]

**Comment by dbason** [comment]
LGTM.  Lets wait for the other PRs and then rerun the build

File: pkg/demo/services.go [hunk 1]
```diff
@@ -6,15 +6,11 @@ import (
 	loggingv1beta1 "github.com/banzaicloud/logging-operator/pkg/sdk/api/v1beta1"
 	"github.com/banzaicloud/logging-operator/pkg/sdk/model/filter"
 	"github.com/banzaicloud/logging-operator/pkg/sdk/model/output"
-	"github.com/containers/image/v5/docker/reference"
 	demov1alpha1 "github.com/rancher/opni/apis/demo/v1alpha1"
 	appsv1 "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
```
Comment (kralicky) [inline pkg/demo/services.go:12]: For consistency, should rename the v1 import to `corev1`

File: go.mod [hunk 2]
```diff
@@ -4,14 +4,15 @@ go 1.16
...
+	github.com/Microsoft/hcsshim v0.8.16 // indirect
 	github.com/banzaicloud/logging-operator/pkg/sdk v0.7.2
-	github.com/containers/image/v5 v5.12.0
 	github.com/elastic/go-elasticsearch/v7 v7.13.1
 	github.com/go-logr/logr v0.4.0
 	github.com/k3s-io/helm-controller v0.10.0
 	github.com/longhorn/upgrade-responder v0.1.2-0.20210521005936-d72e5ddbc541
 	github.com/onsi/ginkgo v1.16.4
 	github.com/onsi/gomega v1.13.0
+	github.com/opencontainers/image-spec v1.0.2-0.20190823105129-775207bd45b6 // indirect
```
Comment (kralicky) [inline go.mod:15]: These two indirect dependencies are not needed, but they are not removed automatically. You will have to remove them manually from go.mod, then run `go mod tidy`.

**Review by kralicky — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (dbason) [reply go.mod:15]: As discussed this is needed for the e2e tests

**Review by kralicky — APPROVED** [review]

**Pull request merged**